build_projects/shared-build-targets-utils/Publishing/AzurePublisher.cs

```diff
@@ -134,11 +146,59 @@ public IEnumerable<string> ListBlobs(string virtualDirectory)
                 }
             }
 
+            ResetLeaseRenewalTaskState();
+
             throw new Exception($"Unable to acquire lease on {blob}");
         }
 
+        private void ResetLeaseRenewalTaskState()
+        {
+            // Cancel the lease renewal task if it was created
+            if (_leaseRenewalTask != null)
+            {
+                _cancellationTokenSource.Cancel();
+                _leaseRenewalTask = null;
+            }
+        }
+
+        private static void AutoRenewLease(AzurePublisher instance, string blob, string leaseId)
```

Comment:
The only difference I see is `instance.` instead of an implicit `this.`--what does this prevent the implementation from assuming? (Willing to drop this because it's not related to this fix, but I'm very curious.)